cylc/flow/task_job_mgr.py

```diff
@@ -288,16 +288,18 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
 
                     # Get another platform, if task config platform is a group
                     use_next_platform_in_group = False
-                    try:
-                        platform = get_platform(
-                            itask.tdef.rtconfig['platform'],
```

Comment:
Problem was `itask.tdef.rtconfig['platform']` is `None` so `get_platform()` fell back to the localhost platform